### PR TITLE
Lapackpp onemkl

### PR DESCRIPTION
**Pull request by ayarkhan**
Testing lapackpp on jlse-arcticus.
module load -f oneapi/eng-compiler/2022.12.30.002
qsub -t 60 -n 1 -q arcticus -I
cd ~/icl/slate-dev/lapackpp/test/
./run_tests.py --quick

Before this PR, on Intel-jlse-arcticus
10 routines FAILED: dev-getrf, dev-potrf, dev-geqrf, hpev, hpevx, hpevx, hptrd, hpgv, hpgvx, geev
Fri Jan  6 17:36:33 2023

After this PR, on arcticus the device routines (dev-getrf, dev-potrf, dev-geqrf) now pass.
6 routines FAILED: hpev, hpevx, hpevx, hpgv, hpgvx, geev
Sun Jan  8 00:50:44 2023

### CODE REVIEW

**Comment by mgates3** [comment]
For future reference, the preference on github is to fork the repo (e.g., https://github.com/mgates3/lapackpp) and create the PR from your fork to the main repo. This is safer for rebasing branches. On bitbucket we had to push all the branches into one repo because of Jenkins limitations.

File: src/onemkl/onemkl_potrf.cc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+#if defined(LAPACK_HAVE_ONEMKL)
+
+#include "onemkl_common.hh"
+
+//==============================================================================
+// todo: put into BLAS++ header somewhere.
+
+namespace blas {
+namespace device {
```
Comment (mgates3) [inline src/onemkl/onemkl_potrf.cc:16]: This changed from `device` to `internal` in BLAS++. I'm fixing the CUDA and ROCm files.

**Comment by mgates3** [comment]
@ayarkhan Can you add the output for the failed routines (hpev, hpevx, hpevx, hpgv, hpgvx, geev). These are all eigenvalue routines: Hermitian packed (hpev), Hermitian packed generalized (hpgv), general (geev). Maybe it's just a tolerance issue. It looks like they do have more robust backwards error checks, not just comparing to LAPACKE.

File: src/onemkl/onemkl_potrf.cc (hunk 1 above)
Comment (ayarkhan) [reply src/onemkl/onemkl_potrf.cc]: I fixed that in the recent commit.
https://github.com/icl-utk-edu/lapackpp/pull/17/commits/a74fd096da6fb51d055b7c6b1eae443ba6e65dd3

**Comment by ayarkhan** [comment]
I have updated the configure script for OneAPI (I was previously using a manual make.inc file).
make distclean followed by make now builds fine on OneAPI/arcticus.
Tests all pass.... including the tests that were failing before (hpev, hpevx, hpevx, hptrd, hpgv, hpgvx, geev).

module load oneapi/eng-compiler/2022.10.15.005
qsub -t 60 -n 1 -q arcticus -I
cd  slate-dev/lapackpp/test/
./run_tests.py 
All routines passed.
Elapsed 1075.53 sec
Thu Jan 12 18:37:17 2023

./run_tests.py --device
All routines passed.
Elapsed 56.45 sec
Thu Jan 12 18:54:35 2023

File: src/onemkl/onemkl_geqrf.cc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    // for cuda, rocm, call set_device; for oneapi, do nothing.
+    blas::internal_set_device( queue.device() );
+
+    int64_t lwork = dev_work_size / sizeof(scalar_t);
+    blas_dev_call(
+        oneapi::mkl::lapack::geqrf(
+            solver, m, n, dA, ldda, dtau, (scalar_t*)dev_work, lwork ));
+
+    // todo: default info returned
+    blas::device_memset( dev_info, 0, 1, queue );
```
Comment (mgates3) [inline src/onemkl/onemkl_geqrf.cc:58]: See issue #18. Let's leave as-is for now. I'm not sure the best way to deal with this.

**Review by mgates3 — APPROVED** [review]

**Pull request merged**